app/Makefile

```diff
@@ -114,13 +114,16 @@ check: format-check lint test
 #########################
 
 # Docker starts the image for the DB but it's not quite
-# fully ready, so add a 5 second sleep so upgrade doesn't
-# fail because the DB hasn't started yet.
-init-db: start-db sleep-5 db-migrate
+# ready to accept connections so we add a brief wait script
+init-db: start-db wait-for-db db-migrate
 
 start-db:
 	docker-compose up --detach main-db
```

Comment:
Now that I look at this with fresh eyes, is there any reason not to put the `wait-for-local-db.sh` call as part of `start-db`? It seems potentially safest that _any_ time we call start-db that we wait for the db to finish booting up before proceeding. Something like


```suggestion
# ready to accept connections so we add a brief wait script
init-db: start-db db-migrate

start-db:
	docker-compose up --detach main-db
	./wait-for-local-db.sh
```

Reply:
I personally like having each command do just one thing, can help with testing/debugging it. For example, when setting this particular script up, I ran the individual components one-by-one to verify the behavior worked even if it was run repeatedly on an already running DB / no DB exists.

Reply:
In that case, what about making `wait-for-db` a dependency of `start-db` e.g.

```
start-db: wait-for-db
  ...
```

Reply:
That would still not let you run the commands individually via the Make command. `init-db` is the one that groups all of the more specific ones together as it is organized right now. We could make an intermediate command and have it like:

```Make
init-db: start-and-wait-for-db db-migrate

# come up with a better name
start-and-wait-for-db: start-db wait-for-db

start-db:
	docker-compose up --detach main-db

wait-for-db:
	./bin/wait-for-local-db.sh

```
But if we add too many it might get difficult to follow.

Reply:
I'm not convinced it's that important to be able to run everything individually. If really needed, it should be possible to just run the shell command directly like `./bin/wait-for-local-db.sh`

If we just kept it simple as:

```
init-db: start-db db-migrate

start-db:
  docker-compose up --detach main-db
  ./bin/wait-for-local-db.sh
```

it's still possible to test all the individual scenarios when testing changes to the Makefile:

common scenarios that devs will use on a regular basis should have make targets

- get everything set up: `make init-db`
- starting the db but not running migrations: `make start-db` (should wait for db to finish booting, which is the common case that devs care about)
- just running migrations: `make db-migrate`

uncommon scenarios that are only relevant when making changes to the make targets themselves don't need special make targets since we can test out the commands manually:

- starting without waiting: `docker-compose up --detach main-db`
- just waiting: `./bin/wait-for-local-db.sh`

Reply:
That makes sense, switched it to what you're suggesting.